src/gui/vector/qgsvectorlayersavestyledialog.cpp

```diff
@@ -43,19 +43,23 @@ QgsVectorLayerSaveStyleDialog::QgsVectorLayerSaveStyleDialog( QgsVectorLayer *la
   const QString myLastUsedDir = settings.value( QStringLiteral( "style/lastStyleDir" ), QDir::homePath() ).toString();
 
   // save style type combobox
-  connect( mStyleTypeComboBox, qOverload<int>( &QComboBox::currentIndexChanged ), this, [ = ]( int )
-  {
-    const QgsVectorLayerProperties::StyleType type = currentStyleType();
-    mSaveToFileWidget->setVisible( type != QgsVectorLayerProperties::DB );
-    mSaveToDbWidget->setVisible( type == QgsVectorLayerProperties::DB );
-    mStyleCategoriesListView->setEnabled( type == QgsVectorLayerProperties::QML );
-    mFileWidget->setFilter( type == QgsVectorLayerProperties::QML ? tr( "QGIS Layer Style File (*.qml)" ) : tr( "SLD File (*.sld)" ) );
-    updateSaveButtonState();
-  } );
+  connect( mStyleTypeComboBox, qOverload<int>( &QComboBox::currentIndexChanged ), this, &QgsVectorLayerSaveStyleDialog::styleTypeChanged );
+  connect( mStyleTypeComboBoxMultipleStyles, qOverload<int>( &QComboBox::currentIndexChanged ), this, &QgsVectorLayerSaveStyleDialog::styleTypeChanged );
+
+  // Add items in the combobox which are compatible with one-style save
   mStyleTypeComboBox->addItem( tr( "As QGIS QML Style File" ), QgsVectorLayerProperties::QML );
   mStyleTypeComboBox->addItem( tr( "As SLD Style File" ), QgsVectorLayerProperties::SLD );
+  mStyleTypeComboBox->addItem( tr( "As default in local database" ), QgsVectorLayerProperties::LOCAL );
+
+  // Add items in the combobox which are compatible with multiple-style save
+  mStyleTypeComboBoxMultipleStyles->addItem( tr( "As QGIS QML Style File" ), QgsVectorLayerProperties::QML );
```

Comment:
Wouldn't be better to have a unique combobox which is populated/updated according to whether we have multiple style to save or not ?

Reply:
Yes! I'm not happy with that but I can't find how to update a combobox in an easy way (hide an item for insance).

The other option than hiding a combobox item would be to keep all the choices in the combobox but disable the dialog `OK` button if the option is selected when it should not be.

Reply:
You could have a populate method that call the QComboBox clear() method at the beginning, and then add the appropriate items.

Reply:
Thanks! It has been corrected